src/mappings/utils.ts

```diff
@@ -113,43 +114,55 @@ export async function upsertEraValue(
   };
 }
 
+export async function createIndexer({
+  address,
+  metadata = '',
+  active = true,
+  createdBlock,
+  lastEvent,
+  controller,
+}: CreateIndexerParams): Promise<Indexer> {
+  const indexer = Indexer.create({
+    id: address,
+    metadata: metadata,
+    totalStake: {
+      era: -1,
+      value: BigInt(0).toJSONType(),
+      valueAfter: BigInt(0).toJSONType(),
+    },
+    commission: {
+      era: -1,
+      value: BigInt(0).toJSONType(),
+      valueAfter: BigInt(0).toJSONType(),
+    },
+    active,
+    controller,
+    createdBlock,
+    lastEvent,
+  });
+
+  await indexer.save();
+  return indexer;
+}
+
 export async function updateTotalStake(
   eraManager: EraManager,
   indexerAddress: string,
   amount: bigint,
   operation: keyof typeof operations,
   applyInstantly?: boolean
 ): Promise<void> {
-  let indexer = await Indexer.get(indexerAddress);
+  const indexer = await Indexer.get(indexerAddress);
 
-  if (!indexer) {
-    indexer = Indexer.create({
-      id: indexerAddress,
-      totalStake: await upsertEraValue(
-        eraManager,
-        undefined,
-        amount,
-        operation,
-        applyInstantly
-      ),
-      commission: await upsertEraValue(
-        eraManager,
-        undefined,
-        BigInt(0),
-        operation,
-        applyInstantly
-      ),
-      active: true,
-    });
-  } else {
-    indexer.totalStake = await upsertEraValue(
-      eraManager,
-      indexer.totalStake,
-      amount,
-      operation,
-      applyInstantly
-    );
-  }
+  if (!indexer) return;
```

Comment:
```
 if (!indexer) {
    indexer = Indexer.create({
      id: indexerAddress,
      totalStake: await upsertEraValue(
      .....
```
or throw error

Reply:
Is this needed?, @HZHAIX and I discussed that Indexer only needs to be created in handleCommissionRate and handleRegisterIndexer.

Reply:
handleCommissionRate should be the only case where it comes before the handleRegisterIndexer event.

Reply:
Aha, `addDelegationEvent` also trigger align with them. Just discuss with Cherry, for this part we will report exception and include `assert` action in `reportException`